Source/MQTTnet/Client/Options/MqttClientTlsOptionsBuilder.cs

```diff
@@ -131,6 +137,17 @@ public MqttClientTlsOptionsBuilder WithCipherSuitesPolicy(EncryptionPolicy encry
             _tlsOptions.EncryptionPolicy = encryptionPolicy;
             return this;
         }
+
+        public MqttClientTlsOptionsBuilder WithCaFile(string caFile)
+        {
+            if (_tlsOptions.CertificateValidationHandler != null)
```

Comment:
WithCaFile needs to configure the CertificateValidationHandler, if it was manually assigned this throws an exception.